tensorflow/core/kernels/slice_op.cc

```diff
@@ -205,6 +205,207 @@ class SliceOp : public OpKernel {
   }
 };
 
+#ifdef INTEL_MKL
+template <typename Device, typename T>
+class MklSliceOp : public OpKernel {
+ public:
+  explicit MklSliceOp(OpKernelConstruction* context) : OpKernel(context) {}
+
+  void Compute(OpKernelContext* context) override {
+    TensorShape output_shape;
+    bool is_identity = true;
```

Comment:
I had thought about various implementation choices. 

1) Since we are special casing for 4-D tensor slicing over channel, I thought about instantiating HandleCase function for 4-D. But instantiating templates for member functions require templates for class (SliceOp) to be instantiated as well. And I thought that instantiating Slice for CPU and float type is not ideal, since the optimization is equally effective for types other than float. 

2) Inheriting from SliceOp would work to reduce common code in Compute, but I would need to invoke SliceOp::Compute from MklSliceOp::Compute. Is that advised? If that is fine, I can go ahead with the 2nd option.